kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -446,19 +438,19 @@ TableInfo constructTableInfo(TableId table, com.google.cloud.bigquery.Schema big
     if (intermediateTables) {
       // Shameful hack: make the table ingestion time-partitioned here so that the _PARTITIONTIME
       // pseudocolumn can be queried to filter out rows that are still in the streaming buffer
-      builder.setTimePartitioning(TimePartitioning.of(Type.DAY));
+      setTimePartitioningForDefinition(table, builder);
```

Comment:
I think we can leave this line as-is; with intermediate tables the exact partitioning type shouldn't matter too much. As the comment above details, the only reason we want to partition the table to begin with is to use the `__PARTITIONTIME` pseudocolumn to avoid trying to delete rows that are still in the streaming buffer.